lib/fastlane/plugin/wpmreleasetoolkit/actions/common/publish_release_action.rb

```diff
@@ -0,0 +1,63 @@
+require 'fastlane/action'
+require_relative '../../helper/github_helper'
+
+module Fastlane
+  module Actions
+    class PublishReleaseAction < Action
+      def self.run(params)
+        repository = params[:repository]
+        name = params[:name]
+        prerelease = params[:prerelease]
+
+        github_helper = Fastlane::Helper::GithubHelper.new(github_token: params[:github_token])
+
+        url = github_helper.publish_release(
+          repository: repository,
+          name: name,
+          prerelease: prerelease
+        )
+        UI.success("Successfully published GitHub Release #{name}. You can see it at '#{url}'")
+        url
+      end
+
+      def self.description
+        'Publishes an existing GitHub Release still in draft mode'
+      end
+
+      def self.authors
+        ['Automattic']
+      end
+
+      def self.return_value
+        'The URL of the published GitHub Release'
+      end
+
+      def self.details
+        'Publishes an existing GitHub Release still in draft mode'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :repository,
+                                       description: 'The slug (`<org>/<repo>`) of the GitHub repository we want to create the release on',
+                                       optional: false,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :name,
+                                       description: 'The name of the release',
+                                       optional: false,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :prerelease,
+                                       description: 'True if this is a pre-release',
+                                       optional: true,
+                                       default_value: false,
```

Comment:
❓ I'm wondering if we would want to provide a way to keep that value unchanged?

Especially because with this having a `default_value` and the `prerelease` being explicitly passed to the API call in the helper, that means that if we ask to `publish_release("12.3-rc-2")` of a _beta_ GitHub Release draft (which had "This is a pre-release" checkbox checked at the time), without explicitly passing `prerelease: true` to that call site, this will change the GitHub Release from initially being marked as a pre-release when it was created as Draft to being a final release when published.

Sadly, even if we set to `optional: true` but with `default_value: nil`, _fastlane_ will make `options[:prerelease]` return `false` (and not `nil`) if we call the action without the parameter. But I just checked and if we set `default_value: :unchanged`, then `fastlane` is happy to have `options[:prerelease]` return that `:unchanged` symbol (despite `type: Boolean`—it doesn't seem to doo any type-check when using the default value). So that means that we could use that to our advantage here, and in `self.run` do something like:

```ruby
        url = github_helper.publish_release(
          repository: repository,
          name: name,
          prerelease: (prerelease == :unchanged) ? nil : prerelease
        )
```

Reply:
Good idea 👍 updated on c283c5e7cf4e578f864d86a5f89ce2ae4f8c9120